File: src/main/java/com/bajdcc/LALR1/grammar/runtime/RuntimeInstBinary.java
package com.bajdcc.LALR1.grammar.runtime;

/**
 * 【中间代码】二元操作符
 *
 * @author bajdcc
 */
public class RuntimeInstBinary extends RuntimeInstUnary {

	public int op2;

	public RuntimeInstBinary(RuntimeInst inst, int op1, int op2) {
		super(inst, op1);
		this.op2 = op2;
	}

	@Override
	public int getAdvanceLength() {
		return 4 + super.getAdvanceLength();
	}

	@Override
	public void gen(ICodegenByteWriter writer) {
		super.gen(writer);
		writer.genOp(op2);
	}

	@Override
	public String toString(String delim) {
		return super.toString(delim) + delim + op2;
	}

	@Override
	public String toString() {
		return toString(" ");
	}
}

File: src/main/kotlin/com/bajdcc/LALR1/grammar/runtime/RuntimeInstBinary.kt
package com.bajdcc.LALR1.grammar.runtime

/**
 * 【中间代码】二元操作符
 *
 * @author bajdcc
 */
class RuntimeInstBinary(inst: RuntimeInst, op1: Int, var op2: Int) : RuntimeInstUnary(inst, op1) {

    override val advanceLength: Int
        get() = 4 + super.advanceLength

    override fun gen(writer: ICodegenByteWriter) {
        super.gen(writer)
        writer.genOp(op2)
    }

    override fun toString(delim: String): String {
        return super.toString(delim) + delim + op2
    }

    override fun toString(): String {
        return toString(" ")
    }
}
